bbl_screen-patch/patches/printerui/qml/settings/ScreenLock.qml

```diff
@@ -5,37 +5,40 @@ import QtQml 2.12
 import UIBase 1.0
 import Printer 1.0
 import X1PlusNative 1.0
-
+import "../X1Plus.js" as X1Plus
 import "qrc:/uibase/qml/widgets"
 import ".."
 
 Rectangle {
+    id: top
+    width:1180
+    height:720
     property var locked: false
     property var isEnteringPasscode: false
     property var passcode: DeviceManager.getSetting("cfw_passcode", "")
     property var locktype: DeviceManager.getSetting("cfw_locktype", 0)
     /* 0 = screensaver only, 1 = swipe to unlock, 2 = passcode */
-    property var lockImage: X1PlusNative.getenv("EMULATION_WORKAROUNDS") + DeviceManager.getSetting("cfw_lockscreen_image", '/mnt/sdcard/x1plus/lockscreen.png')
+    property var lockImagePath: X1PlusNative.getenv("EMULATION_WORKAROUNDS") + DeviceManager.getSetting("cfw_lockscreen_image", '/mnt/sdcard/x1plus/lockscreen.png')
+    property var lockImage: X1Plus.fileExists(lockImagePath) ? lockImagePath : "qrc:/printerui/image/lockscreen.png"
     color: Colors.gray_800
     visible: locked && locktype != 0
     
     property var customText: null
-    
-    id: top
-    ZRoundedImage {
-            id: lockimg
-            anchors.fill: top
-            cornerRadius: parent.radius
-            originSource: "file://" + lockImage
-            visible: imgExists(lockImage)
+
+    Binding on passcode { //possibly a redundant property
+        value: numberPad.number
+        when: numberPad.target == lockText
     }
```

Comment:
I'm not sure I understand this?  this seems to override the passcode entirely such that any passcode will unlock the printer?